### PR TITLE
Split functions and other smaller improvements for 1.6.3

### PR DESCRIPTION
**Pull request by dmpe**
### Merge after code review. This is a tentative changelog:

- [x] Bump to 1.6.x (future release), which as described in #50 should have better error handling and splitted functions (`ErrorHandling.R`).
- [x] 4x4 is now improved as one test in the `test-fourByFour` - the `isFourByFour("4334-!gaj")` - would fail (on my PC) with RSocrata CRAN version. 
- [x] `RSocrata.R` is renamed into `returnData.R`, the same is with the `getResponse` -> `checkResponse` (this function is now quite unnecessary though)
- [x] Added 3rd. regex for the http://dev.socrata.com/docs/datatypes/floating_timestamp.html
- [x] Added some examples & tests and some general formatting according to style guide
- [x] Added vignette (with geo example & move all examples out of the readme file)
   - Tests on Travis take long, but it is ok
   - other small changes to `.md` files
 

### Before merging/releasing on CRAN, this can/needs to be done:
- [x] delete note.md.
- (https://github.com/Chicago/RSocrata/issues/19) -> to consider ?
- [ ] Finish one unit test in `testDatetime.R` -> https://github.com/Chicago/RSocrata/issues/27
- [x] Update News file
- [ ] Version 1.6.3. or 1.6.2 ?
- [ ] Reconsider `@export`ing `fieldName` and `isFourByFour` functions. Shouldn't they be invisible to the user, i.e. `@noRd`

----
### Current File Structure in `R` folder:

**`returnData.R`** has 4 functions:
- read.socrata
- checkResponse
- getContentAsDataFrame
- getSodaTypes

**`utils.R`** has 2 functions:
- fieldName
- posixify
- isFourByFour

**`validateURL.R`** has 1 function:
- validateUrl

**`listDatasets.R`** has 1 function:
- ls.socrata

**`errorHandling.R`** has 1 function:
- errorHandling

### CODE REVIEW

File: note.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Hi @tomschenkjr, 
+Can I just ask something ? We have a function called posixify which "Converts Socrata calendar_date string to POSIX". Can you give me an example of a such string that is then converted to POSIX ?
+
```
Comment (dmpe) [inline note.md]: already found (see examples)

**Comment by dmpe** [comment]
https://stackoverflow.com/questions/10699511/difference-between-as-posixct-as-posixlt-and-strptime-for-converting-character-v for #8

File: tests/testthat/test-dateTime.R [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+
+test_that("posixify new Floating Timestamp format", {
+  dt <- posixify("2014-10-13T23:25:47")
+  expect_equal("POSIXlt", class(dt)[1], label="first data type of a date")
+  expect_equal(2014, dt$year + 1900, label="year")
+  expect_equal(25, dt$min, label="minutes")
+  expect_equal(47, dt$sec, label="seconds")
+})
+
+test_that("NA datetime in source", {
```
Comment (dmpe) [inline tests/testthat/test-dateTime.R]: do somehting with the speed and dplyr

**Comment by dmpe** [comment]
Hello @tomschenkjr & @geneorama, 

Above you can find what changes / (smaller) improvements I have made to the package. Can you please take a look at that, over here https://github.com/Chicago/RSocrata/pull/53/files. I went a bit over the `sprint7` branch, but did rather very smaller changes to the overall code. 

Thank you.

**Comment by tomschenkjr** [comment]
Fantastic. I'll take a look at it. Might be a couple of days before I get a chance.

**Comment by tomschenkjr** [comment]
Accepting pull request. Can you separate-out the additional commits from outstanding pull requests, though? It's getting a bit treacherous trying to track the status of the [CI status](https://travis-ci.org/Chicago/RSocrata/pull_requests) on this one.

**Pull request merged**